functions/test.js

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019, Google, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import test from 'ava'
+import { shiritoriApp } from '.'
+import sinon from 'sinon'
+import admin from 'firebase-admin'
+
+const corpus = require('./corpus.json')
+
+test.before(t => {
+  sinon.stub(Math, 'random').returns(0)
+})
+
+test.serial('welcome', async t => {
+  t.snapshot((await shiritoriApp({'inputs': [
+    {
+      'intent': 'actions.intent.MAIN'
+    }
+  ]}, {})).body)
+})
+
+test.serial('game: next', async t => {
+  sinon.stub(admin, 'database').get(() => () => fakeDb(corpus))
+  t.snapshot((await shiritoriApp({'inputs': [
+    {
+      'intent': 'actions.intent.TEXT',
+      'rawInputs': [
+        {
+          'inputType': 'VOICE',
+          'query': 'とんかつ'
+        }
+      ]
+    }
+  ]}, {})).body)
+})
+
+test.serial('game: loose', async t => {
+  sinon.stub(admin, 'database').get(() => () => fakeDb(corpus))
+  t.snapshot((await shiritoriApp({'inputs': [
+    {
+      'intent': 'actions.intent.TEXT',
+      'rawInputs': [
+        {
+          'inputType': 'VOICE',
+          'query': '座布団'
+        }
+      ]
+    }
+  ]}, {})).body)
+})
+
+test.serial('game: win', async t => {
+  sinon.stub(admin, 'database').get(() => () => fakeDb({'つ':[]}))
```

Comment:
Should we bring the admin.database to its original value after test is done? From stubs docs, 

"var stub = sinon.stub(object, "method"); Replaces object.method with a stub function. An exception is thrown if the property is not already a function.

The original function can be restored by calling object.method.restore(); (or stub.restore();)."[[1](https://sinonjs.org/releases/latest/stubs/)]

Reply:
Done